processor/sumologicprocessor/README.md

```diff
@@ -0,0 +1,320 @@
+| `k8s.node.name`           | `node`              |
+| `k8s.service.name`        | `service`           |
+| `k8s.pod.hostname`        | `host`              |
+| `k8s.pod.name`            | `pod`               |
+| `k8s.pod.uid`             | `pod_id`            |
+| `k8s.replicaset.name`     | `replicaset`        |
+| `k8s.statefulset.name`    | `statefulset`       |
+| `service.name`            | `service`           |
+| `log.file.path_resolved`  | `_sourceName`       |
+
+### Nesting attributes
+
+Nesting attributes allows to change the structure of attributes (both resource level and record level attributes)
+basing on their keys by nesting attributes with common paths into maps.
+Common path is defined as a common prefix that consists of strings separated by a given separator string (by default a dot - `.`)
+and end with that separator, for example: if the separator is `.`,
+then `xyz` is a common prefix path for `xyz.abc.qwe` and `xyz.foo`,
+but `sumo` **is not** a prefix path for neither `sumologic.foo.baz` nor `sumologic.bar.baz`.
+
+#### Example
+
+The following set of attributes:
+
+```json
+{
+  "kubernetes.container_name": "xyz",
+  "kubernetes.host.name": "the host",
+  "kubernetes.host.address": "127.0.0.1",
+  "kubernetes.namespace_name": "sumologic",
+  "another_attr": "42"
+}
+```
+
+Should be translated into such set:
+
+```json
+{
+  "kubernetes": {
+    "container_name": "xyz",
+    "namespace_name": "sumologic",
+    "host": {
+      "name": "the host",
+      "address": "127.0.0.1"
+    }
+  },
+ "another_attr": "42"
+}
+```
+
+#### Allowlist and denylist
+
+Properties `include` and `exclude` in the config allow to define list of prefixes that are allowed or not allowed to be nested.
+
+For example, with the following config:
+
+```yaml
+nest_attributes:
+  enabled: true
+  include: ["kubernetes.host."]
+  exclude: ["kubernetes.host.naming"]
+```
+
+and following input:
+
+```json
+{
+"kubernetes.container_name": "xyz",
+"kubernetes.host.name": "the host",
+"kubernetes.host.naming_convention": "random",
+"kubernetes.host.address": "127.0.0.1",
+"kubernetes.namespace_name": "sumologic",
+}
+```
+
+The output is:
+
+```json
+{
+"kubernetes.container_name": "xyz",
+"kubernetes": {
+  "host": {
+    "name": "the host",
+    "address": "127.0.0.1"
+    }
+  },
+"kubernetes.host.naming_convention": "random",
+"kubernetes.namespace_name": "sumologic",
+}
+```
+
+#### Known issues
+
+This feature has undefined behavior when in input there are various keys that will be mapped to the same nested structure.
+For example, given the following attributes:
+
+```json
+{
+  "a.b.c.": "d",
+  "a": {
+    "b": {
+      "c": "e"
+    }
+  }
+}
+```
+
+It is not possible to predict, what will be the result. It will have the following structure:
+
+```json
+{
+  "a": {
+    "b": {
+      "c": ...
+    }
+  }
+}
+```
+
+However, it is not possible to know a priori if the value under key `c` will be equal to `d` or `e`.
```

Comment:
TIL

Reply:
Just a minor note, it may be worth changing "a priori" to "beforehand" considering the global audience of the Collector who may not be familiar with common Latin phrases.